extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/DifferenceWithinRangeScrutinizer.java

```diff
@@ -0,0 +1,53 @@
+package org.openrefine.wikidata.qa.scrutinizers;
+
+import org.openrefine.wikidata.qa.QAWarning;
+import org.openrefine.wikidata.updates.ItemUpdate;
+import org.wikidata.wdtk.datamodel.interfaces.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DifferenceWithinRangeScrutinizer extends EditScrutinizer {
+
+    public static final String type = "difference-of-the-properties-is-not-within-the-specified-range";
+
+    @Override
+    public void scrutinize(ItemUpdate update) {
+        Map<PropertyIdValue, Value> propertyIdValueValueMap = new HashMap<>();
+        for (Statement statement : update.getAddedStatements()){
+            PropertyIdValue pid = statement.getClaim().getMainSnak().getPropertyId();
+            Value value = statement.getClaim().getMainSnak().getValue();
+            propertyIdValueValueMap.put(pid, value);
+        }
+
+        for(PropertyIdValue propertyId : propertyIdValueValueMap.keySet()){
+            if (_fetcher.hasDiffWithinRange(propertyId)){
+                PropertyIdValue lowerPropertyId = _fetcher.getLowerPropertyId(propertyId);
+                QuantityValue minRangeValue = _fetcher.getMinimumValue(propertyId);
+                QuantityValue maxRangeValue = _fetcher.getMaximumValue(propertyId);
+
+                if (propertyIdValueValueMap.containsKey(lowerPropertyId)){
+                    Value startingValue = propertyIdValueValueMap.get(lowerPropertyId);
+                    Value endingValue = propertyIdValueValueMap.get(propertyId);
+
+                    if (startingValue instanceof TimeValue && endingValue instanceof TimeValue){
```

Comment:
This constraint might also be used on quantities: it would be good to support that case too.

Reply:
I took a look at the [uses](https://www.wikidata.org/w/index.php?title=Special:WhatLinksHere&target=Q21510854&namespace=120) of this constraint, and all of them are for date & year. Therefore, I made this specific condition for TimeValue only. 

If there are chances to use this constraint on other quantities then should I make a different class extending this one for TimeValue and so on for other quantities?

Reply:
Good question. Perhaps it is worth checking the code of WikibaseQualityConstraints to see how it was implemented there (perhaps quantities are not supported indeed!)
https://github.com/wikimedia/mediawiki-extensions-WikibaseQualityConstraints

Reply:
I checked their implementation, quantities are not really supported over there too

Reply:
Apparently they are actually supported! :)

Reply:
Yes, they do.  I'm trying to implement something similar to rangeCheckerHelper and UnitsConverter. Once it gets implemented very well and validated I'll update the PR. implementing these kinds of classes will provide a good help in range constraints too. :)